cli/internal/publish/images/images.go

```diff
@@ -39,6 +41,25 @@ type listKey struct {
 	name, sum string
 }
 
+func QuickContentType(filename string) (string, error) {
+	filebytes := make([]byte, 512)
+
+	fp, err := os.Open(filename)
+	if err != nil {
+		return "", fmt.Errorf("failed to open file: %w", err)
+	}
+	defer fp.Close()
+	if _, err := fp.Read(filebytes); err != nil {
+		return "", fmt.Errorf("failed to read file: %w", err)
+	}
+
+	contentType := mime.TypeByExtension(filepath.Ext(filename))
+	if contentType == "" {
+		contentType = http.DetectContentType(filebytes)
```

Comment:
If this doesn't detect a content type, should it error?

Reply:
From the godoc:
>  if it cannot determine a more specific one, it returns "application/octet-stream".